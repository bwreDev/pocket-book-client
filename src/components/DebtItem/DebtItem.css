.debt-item {
  text-transform: capitalize;
  display: flex;
  flex-direction: column;
  text-align: center;
  margin-bottom: 0;
  margin-top: 0;
}

.debt-amount,
.debt-description {
  margin: 0;
}

.debt-post-time {
  font-size: smaller;
}
