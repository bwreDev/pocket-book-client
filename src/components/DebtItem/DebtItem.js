import React from 'react';
import moment from 'moment';
import './DebtItem.css';

export default function DebtItem(props) {
  const cleanDate = moment(props.input.date_added).format(
    'MMM Do YY, h:mm a'
  );

  return (
    <div className='debt-item'>
      <p className='debt-amount'>${props.input.amount}</p>
      <p className='debt-description'>{props.input.content}</p>
      <span className='debt-post-time'>
        {props.input.date_added ? cleanDate : ''}
      </span>
    </div>
  );
}

DebtItem.defaultProps = {
  input: {},
};
