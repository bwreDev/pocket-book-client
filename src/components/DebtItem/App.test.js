import React from 'react';
import ReactDOM from 'react-dom';
import DebtItem from './DebtItem';
import renderer from 'react-test-renderer';

it('should render without error', () => {
  const div = document.createElement('div');
  ReactDOM.render(<DebtItem />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer.create(<DebtItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
