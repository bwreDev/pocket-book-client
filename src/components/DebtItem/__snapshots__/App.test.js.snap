// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <p
    className="debt-amount"
  >
    $
  </p>,
  <p
    className="debt-description"
  >
     - 
  </p>,
  <span
    className="debt-post-time"
  >
    
  </span>,
]
`;
